feat: add routes details and domain metadata helper functions

MakMuftic opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Related Issue Or Context
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Otherwise, describe context and motivation for change herre -->

Closes: #334 #344 

## How Has This Been Tested? Testing details.
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

mpetrunic commented on packages/sdk/src/utils.ts:
```diff
@@ -50,3 +58,71 @@ export async function getTransferStatusData(
+ */
+export async function getRoutes(
+  environment: Environment,
+  sourceChainId: number,
```
nvm found hack

mpetrunic commented on packages/sdk/src/types/types.ts:
```diff
@@ -91,3 +91,25 @@ export type TransferStatusResponse = {
+};
+
+export type EnvironmentMetadata = {
+  // domainID -> DomainMetadata
```
does it also work for resourceId?

MakMuftic replied:
This will be a separate PR

mpetrunic reviewed (commented):
You should probably open issue to use this in SDK, as right now if route is unsupported this is error from SDK:
![image](https://github.com/sygmaprotocol/sygma-sdk/assets/8836210/b5b42836-a727-4d99-8122-3b7a4e4c653b)

😅

MakMuftic commented:
## New updates
- removed `axios` as dependency
  - we were using both axios and fetch in codebase, so refactored it to use fetch
- added `isTransferValid` function to BaseAssetTransfer

mpetrunic commented on packages/sdk/package.json:
```diff
@@ -84,8 +84,7 @@
     "@polkadot/util-crypto": "12.2.1",
-    "axios": "^1.4.0",
-    "cross-fetch": "^3.1.5",
+    "cross-fetch": "^4.0.0",
```
https://github.com/sygmaprotocol/sygma-sdk/issues/344 😛
